### ISSUE TITLE
MyWebsocket need handle ping pong

### ISSUE THREAD

**Issue by 1AlexFix1** [dart-lang/sdk#36728]
Hello, We need to know when was ping and pong.
I also looked at the WebSocket class from the dart.io package and could not send where pingInterval is used. The _WebSocketImpl implementation is closed.
The problem is that the connection often disappears in GoingAway if the ping interval is below 10 seconds

**Comment by devoncarew** [comment]
It looks like this is a request to open up more of the websocket implementation as API.

**Comment by devoncarew** [comment]
@a-siva, which engineer maintains `dart:io` now?

**Comment by athomas** [comment]
That's @sortie now.

**Comment by 1AlexFix1** [comment]
> It looks like this is a request to open up more of the websocket implementation as API.

We checked how ping-pong works, and from the documentation it was not very clear, at the moment when pintInterval works as follows, sends ping requests waiting for pong, and when pong arrives, ping will be sent only after the remaining period of time. As far as I know, the usual ping-pong behavior is ping once a second, and if the pong does not return during pingInterval, the socket closes

**Comment by StanevDev** [comment]
I'm develop mobile flutter app witch use websocket, and have issue with long-live connections,  before 20 minutes connection socket don't send messages to backend & don't receive them.
pingInterval setting did not help and I can't debug private implementation of ping-pong.

**Comment by sortie** [comment]
@zichangg Can you confirm if this is a duplicate of the issue you recently fixed?

**Comment by zichangg** [comment]
@StanevDev Could you please verify what version of dart you are using?

**Comment by sortie** [comment]
@zichangg Note this is an old issue from long before your fixes. I'm trying to determine whether it is a duplicate issue and whether it can be closed as such. If you don't think we have enough information to diagnose it, then you can add the needs-info label and the issue will be auto-closed if no-one responds for a while.

**Comment by zichangg** [comment]
@sortie From the description, I think this is a duplicate.
My fix went in about 2 months ago, but @StanevDev 's comment is very recent, it is probably a different issue.

**Comment by StanevDev** [comment]
@zichangg Dart version is 2.8.1. I found out that problem is in background app state.

**Comment by zichangg** [comment]
@StanevDev Sounds good. I'll close the issue.

**Comment by sportmachine** [comment]
can someone give an example? the example in test are just the basic. while so many difference code are so diverse. I have tried many, most don't work as what Tomcat did.
